### DISCUSSION TITLE
Pulling new metadata from duplicates during import

### DISCUSSION THREAD

**Discussion by tilusnet** [photoprism/photoprism#4329] [Ideas]
Hello,

I assume Photoprism does not enrich metadata like keywords from photos identified as duplicates during import.
If I am correct, I think this could be improved.

Use case:
I have two copies for a set of photos, let's call these S1 and S1':
* S1 has associated metadata sidecar files, and sits under meaningful folder hierarchies. This will result in richer keyword generation during import.
* S1' is a dump of S1, just the files, but identical to those in S1 (hash matches). Importing these results in poorer keywords generation.

Two scenarios can happen:
(1) If Photoprism finds the S1' set first, it imports them without the rich metadata. When it finds the S1 set, it skips them altogether.
(2) If Photoprism finds the S1 set first, it imports them with rich metadata.

I advocate for scenario (2) to happen, regardless of scanning order as this would result in richer metadata.
Is this possible to implement?

**Comment by tilusnet** [comment]
This issue now seems more serious than I originally thought: at least half of my collection's XMP files seem to have been ignored during a 4 day import. Just because the importer seems to have found the "exported" variants first, which did not have the XMP files next to them.

If this makes sense, I will draft a bug report.

IMO the importing end results should be idempotent regarding the metadata collected, it should not depend on the parsing order of the imported content.

**Reply by lastzero** [reply]
If you're using the optional import functionality, then there won't be any duplicates in your library in the first place. This is as designed and documented, thus not a bug. Checking for the existence of additional / alternate XMP files next to duplicates would seem possible, but that would be new functionality, thus a feature request. In this case you might additionally need a way to define priorities and a write / modify strategy once we write data back to those files. So the feature might become highly complex and could require weeks / months of development and testing.
